core/web/transfer_controller.go

```diff
@@ -40,9 +41,63 @@ func (tc *TransfersController) Create(c *gin.Context) {
 		return
 	}
 
-	db := tc.App.GetSqlxDB()
-	q := pg.NewQ(db, tc.App.GetLogger(), tc.App.GetConfig())
-	etx, err := bulletprooftxmanager.SendEther(q, chain.ID(), tr.FromAddress, tr.DestinationAddress, tr.Amount, chain.Config().EvmGasLimitTransfer())
+	if tr.FromAddress == utils.ZeroAddress {
+		jsonAPIError(c, http.StatusUnprocessableEntity, fmt.Errorf("withdrawal source address is missing: %v", tr.FromAddress))
+		return
+	}
+
+	if !tr.AllowHigherAmounts {
```

Comment:
Can we pull the contents into a helper function? Might aid unit testing also

Reply:
Addressed